presto-main/src/main/java/com/facebook/presto/server/security/AuthenticationFilter.java

```diff
@@ -166,4 +218,51 @@ private static void skipRequestBody(HttpServletRequest request)
             copy(inputStream, nullOutputStream());
         }
     }
+
+    public static class CustomHttpServletRequestWrapper
```

Comment:
```suggestion
    private static class ModifiedHttpServletRequest
```

Reply:
All of the public methods on `ModifiedHttpServletRequest` override something from a parent interface or abstract class, so can't you use the most restrictive interface?  Wouldn't that be `HttpServletRequest`?

Reply:
Hi @tdcmeehan 
When I implement the HttpServletRequest interface directly, I must implement all of its abstract methods. 
The HttpServletRequest interface contains many methods, so implementing it directly can be tedious, especially when I only need to override a few methods, such as getHeader, getHeaderNames, and getHeaders.

Reply:
I am not telling you to implement `HttpServletRequest`, I am simply saying that this method can just return the interface `HttpServletRequest` for better [encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)).

Reply:
@tdcmeehan 
I have made the change and committed. Please review the code. 
Thanks.